crates/dojo-lang/src/plugin_test_data/introspect

```diff
@@ -82,6 +106,186 @@ impl Vec2SchemaIntrospection of dojo::database::schema::SchemaIntrospection<Vec2
 
 
 
+#[derive(Serde, Copy, Drop, Introspect)]
+enum PlainEnum {
+    Left: (),
+    Right: (),
+}
+
+impl PlainEnumSchemaIntrospection of dojo::database::schema::SchemaIntrospection<PlainEnum> {
+    #[inline(always)]
+    fn size() -> usize {
+        1
+    }
+
+    #[inline(always)]
+    fn layout(ref layout: Array<u8>) {
+        layout.append(8);
+    }
+
+    #[inline(always)]
+    fn ty() -> dojo::database::schema::Ty {
+        dojo::database::schema::Ty::Enum(
+            dojo::database::schema::Enum {
+                name: 'Direction',
+                attrs: array![].span(),
+                children: array![
+                    (
+                        'Left',
+                        serialize_member_type(@dojo::database::schema::Ty::Tuple(array![].span()))
+                    ),
+                    (
+                        'Right',
+                        serialize_member_type(@dojo::database::schema::Ty::Tuple(array![].span()))
+                    )
+                ]
+                    .span()
+            }
+        )
+    }
+}
+
+
+
+#[derive(Serde, Copy, Drop, Introspect)]
+enum EnumPrimitive {
+    Left: (u16,),
+    Right: (u16,),
+}
+
+impl EnumPrimitiveSchemaIntrospection of dojo::database::schema::SchemaIntrospection<EnumPrimitive> {
+    #[inline(always)]
+    fn size() -> usize {
+        2
+    }
+
+    #[inline(always)]
+    fn layout(ref layout: Array<u8>) {
+        layout.append(8);
+        layout.append(16);
+    }
+
+    #[inline(always)]
+    fn ty() -> dojo::database::schema::Ty {
+        dojo::database::schema::Ty::Enum(
+            dojo::database::schema::Enum {
+                name: 'Direction',
+                attrs: array![].span(),
+                children: array![
+                    (
+                        'Left',
+                        serialize_member_type(
+                            @dojo::database::schema::Ty::Tuple(array!['u16'].span())
```

Comment:
In this case, it should derive a Tuple of Ty::Primitives

Reply:
Now I know why it was nested span...

```rs
                    (
                        'Left',
                        dojo::database::schema::serialize_member_type(
                            @dojo::database::schema::Ty::Tuple(
                                array![
                                    dojo::database::schema::serialize_member_type(
                                        @dojo::database::schema::Ty::Primitive('u8')
                                    ),
                                    dojo::database::schema::serialize_member_type(
                                        @dojo::database::schema::Ty::Primitive('u8')
                                    )
                                ]
                                    .span()
                            )
                        )
                    ),
```